File: EduCollab_benchmark/3_file_level_generated_artifacts/php/educollab_php_vul005/src/repos/UploadRepo.php
<?php

class UploadRepo {
    public function __construct(private PDO $pdo) {}

    public function create(int $courseId, int $uploadedBy, string $originalName, string $storagePath): array {
        $stmt = $this->pdo->prepare('INSERT INTO uploads(course_id,uploaded_by,original_name,storage_path,created_at) VALUES (?,?,?,?,?)');
        $stmt->execute([$courseId, $uploadedBy, $originalName, $storagePath, now_iso()]);
        return $this->getById($courseId, (int)$this->pdo->lastInsertId());
    }

    public function listByCourse(int $courseId): array {
        $stmt = $this->pdo->prepare('SELECT up.*, u.username AS uploaded_by_username FROM uploads up JOIN users u ON u.user_id=up.uploaded_by WHERE up.course_id=? ORDER BY up.upload_id DESC');
        $stmt->execute([$courseId]);
        return $stmt->fetchAll();
    }

    public function getById(int $courseId, int $uploadId): ?array {
        $stmt = $this->pdo->prepare('SELECT up.*, u.username AS uploaded_by_username FROM uploads up JOIN users u ON u.user_id=up.uploaded_by WHERE up.course_id=? AND up.upload_id=?');
        $stmt->execute([$courseId, $uploadId]);
        $row = $stmt->fetch();
        return $row ?: null;
    }

    public function delete(int $courseId, int $uploadId): void {
        $stmt = $this->pdo->prepare('DELETE FROM uploads WHERE course_id=? AND upload_id=?');
        $stmt->execute([$courseId, $uploadId]);
    }
}

File: EduCollab_benchmark/2_requirement_level_generated_artifacts/python/educollab_flask_req05/models/upload_repo.py
from models.db import get_db



def create_upload(course_id, uploaded_by, original_name, storage_path):
    db = get_db()
    cursor = db.execute(
        'INSERT INTO uploads (course_id, uploaded_by, original_name, storage_path) VALUES (?, ?, ?, ?)',
        (course_id, uploaded_by, original_name, storage_path),
    )
    db.commit()
    return get_upload_by_id(course_id, cursor.lastrowid)


def list_uploads(course_id):
    db = get_db()
    return db.execute(
        'SELECT up.upload_id, up.course_id, up.uploaded_by, up.original_name, up.storage_path, up.created_at, '
        'u.username, u.display_name '
        'FROM uploads up JOIN users u ON u.user_id = up.uploaded_by '
        'WHERE up.course_id = ? ORDER BY up.upload_id DESC',
        (course_id,),
    ).fetchall()


def get_upload_by_id(course_id, upload_id):
    db = get_db()
    return db.execute(
        'SELECT up.upload_id, up.course_id, up.uploaded_by, up.original_name, up.storage_path, up.created_at, '
        'u.username, u.display_name '
        'FROM uploads up JOIN users u ON u.user_id = up.uploaded_by '
        'WHERE up.course_id = ? AND up.upload_id = ?',
        (course_id, upload_id),
    ).fetchone()


def delete_upload(course_id, upload_id):
    db = get_db()
    db.execute('DELETE FROM uploads WHERE course_id = ? AND upload_id = ?', (course_id, upload_id))
    db.commit()
